tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,213 @@
+        }
+        $vms->{"$guest-fv"}->{dsturi}       = "xen+ssh://$dst_ip/";
+        $vms->{"$guest-fv"}->{virsh_cmd}    = "virsh migrate --live";
+        $vms->{"$guest-fv"}->{migrate_opts} = [""];                     # leave a anonymous array for future migrate opts extension.
+        $ret                                = script_run("[ -f $images_path/vm-configs/$guest-pv.xml ]");
+        if ($ret == 0) {
+            assert_script_run("virsh define --file $images_path/vm-configs/$guest-pv.xml");
+            $vms->{"$guest-pv"} = {reuse => 1};
+        } else {
+            $vms->{"$guest-pv"} = {
+                name      => "$guest-pv",
+                version   => $guest =~ /-sp/ ? $guest =~ s/.*-(\d+)-sp(\d)/$1.$2/r : $guest =~ s/.*-(\d+)/$1/r,
+                arch      => get_var('ARCH'),
+                v_type    => "-p",
+                location  => $inst_src,
+                path      => "$images_path/$guest-pv.qcow2,size=20",
+                port      => 5907,
+                disk_type => "xvda"
+            };
+            $vms_to_install++;
+        }
+        $vms->{"$guest-pv"}->{dsturi}       = "xen+ssh://$dst_ip/";
+        $vms->{"$guest-pv"}->{virsh_cmd}    = "virsh migrate --live";
+        $vms->{"$guest-pv"}->{migrate_opts} = [""];
+    }
+}
+#Generate a autoyast profile for each vm to be insatlled.
+sub init_vm_profile {
+    my $path = get_required_var('GUEST_AUTOYAST');
+    # Get file from data directory
+    while (my ($vm_name, $vm_info) = each %$vms) {
+        record_info($vm_name);
+        next if $vm_info->{reuse};
+        my $profile = get_test_data($path);
+        $profile =~ s/\{\{ARCH\}\}/$vm_info->{arch}/g;
+        $profile =~ s/\{\{VERSION\}\}/$vm_info->{version}/g;
+        $profile =~ s/\{\{SUT_IP\}\}/$sut_ip/g;
+        $profile =~ s/\{\{GUEST\}\}/$vm_name/g;
+        $profile =~ s/\{\{DISK_TYPE\}\}/$vm_info->{disk_type}/g;
+        $profile =~ s/\{\{IP_PATH\}\}/$nfs_path\/guestip/g;
+        save_tmp_file("$vm_name.xml", $profile);
+        $vm_info->{autoyast} = autoinst_url . "/files/$vm_name.xml";
+        record_info($vm_info->{autoyast});
+        # Copy profile to ulogs directory, so profile is available in job logs
+        make_path('ulogs');
+        copy(hashed_string("$vm_name.xml"), "ulogs/$vm_name.xml");
+        $vm_info->{install_cmd} = "$virt_insatll --name $vm_name $vm_info->{v_type} --disk path=$vm_info->{path} --location $vm_info->{location} -x autoyast=$vm_info->{autoyast} --graphics vnc,port=$vm_info->{port},listen=0.0.0.0";
+        record_info($vm_info->{install_cmd});
+    }
+}
+
+
+sub run {
+    select_console 'root-ssh';
+    $dst_ip   = get_job_autoinst_vars(get_parents->[0])->{SUT_IP};
+    $inst_src = get_test_suite_data()->{location}->{"$guest"};
+    record_info("DST: $dst_ip");
+    record_info("SUT: $sut_ip");
+    assert_script_run("if [ -d $images_path/guestip ]; then rm -rf $images_path/guestip; fi; mkdir $images_path/guestip");
```

Comment:
If you're anyways doing a `rm -rf`, one could skip the previous `if [ -d`. So:

```suggestion
    assert_script_run("rm -rf $images_path/guestip && mkdir -p $images_path/guestip");
```

Reply:
yes, "rm -rf" always return 0.